tests/test_other_pages.py

```diff
@@ -186,3 +186,29 @@ def test_floating_menu(self):
         self.assertGreater(final_top_offset + height/2, scroll_distance)
 
         bp.close()
+
+    def test_autocompletion(self):
+        def get_number_suggestions():
+            suggestions = bp.element_by('id', 'ui-id-1')
+            suggestions.activate_element()
+            suggestions_list = suggestions.element.find_elements_by_tag_name('li')        
+            return len(suggestions_list)
+             
+        cf = Config()
+        cf.http_prefix = 'https://'
+        cf.base_url = 'demoqa.com/autocomplete/'
+        cf.options_list.append("headless")
+        bp = Page(cf)
+        bp.go()
+ 
+        sug_number = []
+        textbox = bp.element_by('id', 'tags')
+        textbox.input_text('a')
+        sug_number.append(get_number_suggestions())
+        textbox.input_text('s')  # needs time to repopulate list
+        time.sleep(0.5)
```

Comment:
@ddrm86 is this sleep required? there's no way to dynamically wait for the suggestions element to populate?

Reply:
Good question...

The thing is all the elements are present all the time: the only change is in the number of items in the list. When there are no suggestions they do not bother removing elements from the list, they just make it invisible. I had no timing problems entering a text for the first time, but when I add the second letter I always have to wait to get the correct number of suggestions.

It looks like the Java version of Selenium has an ExpectedCondition to wait for a number of elements ( https://seleniumhq.github.io/selenium/docs/api/java/org/openqa/selenium/support/ui/ExpectedConditions.html#numberOfElementsToBe-org.openqa.selenium.By-java.lang.Integer- ) but apparently it is not implemented in Python ( https://selenium-python.readthedocs.io/api.html#module-selenium.webdriver.support.expected_conditions )

Reply:
Cool. thank you for working on this